wordpress-mobile/WordPress-iOS
# Blogging Prompts: Add prompt metadata to prompt posts

See: #18429
WPKit PR: https://github.com/wordpress-mobile/WordPressKit-iOS/pull/515

## Description

Adds prompt metadata to posts when answering a prompt in order to track if it has been answered.

## Testing

**Note:** There's an existing caching issue where the prompt data will not update after answering the prompt. A temporary change is needed in `BloggingPromptsService.swift`:

```swift
// BloggingPromptsService.swift

    func todaysPrompt(success: @escaping (BloggingPrompt?) -> Void,
                      failure: @escaping (Error?) -> Void) {
//        guard localTodaysPrompt == nil else {
//            success(localTodaysPrompt)
//            return
//        }

        fetchTodaysPrompt(success: success, failure: failure)
    }
```

To test:
- Enable `bloggingPrompts` feature flag in `FeatureFlag.swift`
- Launch the app and sign in if necessary
- Select a public site
- Navigate to the 'My Site' tab
- Tap on 'Answer Prompt'
- In the editor, publish the post publicly
- After the post is uploaded, trigger the `todaysPrompt` (switch to another blog and then back, this should call the `configure` method on the dashboard card which calls `todaysPrompt`)
- Verify the prompt is marked as 'Answered' and the answer count is updated

## Regression Notes
1. Potential unintended areas of impact
N/A

2. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A

3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:
